src/sentry/integrations/msteams/webhook.py

```diff
@@ -89,25 +94,87 @@ def post(self, request):
+
+        team = channel_data["team"]
 
-                access_token = get_token_data()["access_token"]
+        # TODO: add try/except for request exceptions
+        access_token = get_token_data()["access_token"]
 
-                # need to keep track of the service url since we won't get it later
-                signed_data = {"team_id": team_id, "service_url": data["serviceUrl"]}
+        # need to keep track of the service url since we won't get it later
+        signed_data = {
+            "team_id": team["id"],
+            "team_name": team["name"],
+            "service_url": data["serviceUrl"],
+            "expiration_time": int(time.time()) + INSTALL_EXPIRATION_TIME,
+        }
 
-                # sign the params so this can't be forged
-                signed_params = sign(**signed_data)
+        # sign the params so this can't be forged
+        signed_params = sign(**signed_data)
 
-                # send welcome message to the team
-                client = MsTeamsPreInstallClient(access_token, data["serviceUrl"])
-                client.send_welcome_message(team_id, signed_params)
+        # send welcome message to the team
+        client = MsTeamsPreInstallClient(access_token, data["serviceUrl"])
+        client.send_welcome_message(team["id"], signed_params)
+        return self.respond(status=201)
 
-        return self.respond(status=200)
+    def handle_member_removed(self, request):
+        data = request.data
+        channel_data = data["channelData"]
+        # only care if our bot is the new member removed
+        matches = filter(lambda x: x["id"] == data["recipient"]["id"], data["membersRemoved"])
+        if not matches:
+            return self.respond(status=204)
+
+        team_id = channel_data["team"]["id"]
+
+        try:
+            integration = Integration.objects.get(provider=self.provider, external_id=team_id)
+        except Integration.DoesNotExist:
+            logger.info(
+                "msteams.uninstall.missing-integration", extra={"team_id": team_id},
+            )
+            return self.respond(status=404)
+
+        # no matter how many orgs are using the integration
+        # we have to delete the integration because the auth has been revoked
+        # an app can only be installed once for a team (unless it's deleted and re-installed)
+        # this is different than Vercel, for example, which can have multiple installations
+        # for the same team in Vercel with different auth tokens
+
+        for org in integration.organizations.all():
+            create_audit_entry(
+                request=request,
+                organization=org,
+                target_object=integration.id,
+                event=AuditLogEntryEvent.INTEGRATION_REMOVE,
+                actor_label="Teams User",
+                data={
+                    "provider": integration.provider,
+                    "name": integration.name,
+                    "team_id": team_id,
```

Comment:
where does the `team_id` get used?

Reply:
@MeredithAnya It's just another piece of data we can use if we need to debug something. It just gets stored in the JSON of the row.

Reply:
okays, def have thought of the audit logs as primarily information for the user, not necessarily a debugging tool for us (in terms of storing extra data that doesn't get rendered), so wanted to make sure you weren't relying on it being shown

Reply:
@scefali I see. I don't think any of the information in `data` is actually shown to the user unless I am mistaken: 
![Screen Shot 2020-07-20 at 10 53 43 AM](https://user-images.githubusercontent.com/8533851/87969653-6608e780-ca77-11ea-9f09-d3da8625c508.png)